core/crypto/nike/nike.go

```diff
@@ -78,6 +89,21 @@ type Nike interface {
 	// See also PublicKey's Blind method.
 	Blind(groupMember []byte, blindingFactor []byte) (blindedGroupMember []byte)
```

Comment:
Why is this `[]byte` and not just `groupMember *PublicKey, blindingFactor *PrivateKey` ? Then we can centralize the validation, and it also becomes less unclear how to come up with valid blinding factors.
Looks like it should also return a `*PublicKey` here?

Reply:
We've actually got two Blind ops. The math is the same... both are for the public key blinding but one is called from the Scheme reference and the other is a method of PublicKey. The PublicKey method mutates the public key. Other Scheme method simply returns the blinded value. We have it like this and their arguments take the types they currently take to make it convenient to from this bit of code here:

https://github.com/katzenpost/katzenpost/blob/main/core/sphinx/sphinx.go#L332-L345

Reply:
But @threebithacker I'm open to changing the function signatures but that shouldn't block this pull request. We can do that in another pull request. I mean... @leif already complained that this pull request is 17 files. Wouldn't want to make it even bigger.